Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -1610,6 +1545,35 @@ namespace OpenInfraPlatform {
 				return paths;
 			}
 
+			/*! brief Converts 2D position of the object according to starting position. 
+			*  \param[in] profileDef A pointer to data from IfcProfileDef.
+			*  \param[in] temp_paths A pointer to coordinates of the profile
+			*  \param[out] paths A pointer to be filled with the relevant data.
+			*/
+			void ApplyPosition(const EXPRESSReference<typename IfcEntityTypesT::IfcParameterizedProfileDef>& profileDef,
+				std::vector<std::vector<carve::geom::vector<2>>> temp_paths,
+				std::vector<std::vector<carve::geom::vector<2>>>& paths) const throw(...){
+
+				if (profileDef->Position) {
+					carve::math::Matrix transform = placementConverter->convertIfcPlacement(profileDef->Position.get());
+
+					for (std::vector<carve::geom::vector<2>> path_loop : temp_paths) {
+						for (carve::geom::vector<2> pt : path_loop) {
+							carve::geom::vector<3> pt_3d(carve::geom::VECTOR(pt.x, pt.y, 0));
+							pt_3d = transform * pt_3d;
+							pt.x = pt_3d.x;
+							pt.y = pt_3d.y;
+						}

```

Comment:
Are you trying to overwrite `pt` here? Was it like this also before the changes?

Becase then you would need to initialize the running variable with `&` as: `carve::geom::vector<2> &pt : path_loop`.

Reply:
Resolved: 5ef7ef1